packages/contracts/test/AZTEC/bilateralSwap.js

```diff
@@ -0,0 +1,156 @@
+
+            const { proofData, challenge } = aztec.proof.bilateralSwap.constructBilateralSwap(
+                incorrectTestNoteValues,
+                accounts[0]
+            );
+
+            await exceptions.catchRevert(bilateralSwapContract.validateBilateralSwap(proofData, challenge, t2, {
+                from: accounts[0],
+                gas: 4000000,
+            }));
+        });
+
+        it('Validate failure for incorrect number of input notes', async () => {
+            const makerNoteValues = [10, 20, 30];
+            const takerNoteValues = [10, 20, 30];
+            const incorrectNumberOfNotes = aztec.proof.bilateralSwap.helpers.makeTestNotes(makerNoteValues, takerNoteValues);
+
+            const { proofData, challenge } = aztec.proof.bilateralSwap.constructBilateralSwap(
+                incorrectNumberOfNotes,
+                accounts[0]
+            );
+
+            await exceptions.catchRevert(bilateralSwapContract.validateBilateralSwap(proofData, challenge, t2, {
+                from: accounts[0],
+                gas: 4000000,
+            }));
+        });
+
+        it('Validate failure for a bid note of zero value', async () => {
+            const makerNoteValues = [0, 20];
+            const takerNoteValues = [10, 20];
+            const NotesWithAZero = aztec.proof.bilateralSwap.helpers.makeTestNotes(makerNoteValues, takerNoteValues);
+
+            const { proofData, challenge } = aztec.proof.bilateralSwap.constructBilateralSwap(NotesWithAZero, accounts[0]);
+
+            await exceptions.catchRevert(bilateralSwapContract.validateBilateralSwap(proofData, challenge, t2, {
+                from: accounts[0],
+                gas: 4000000,
+            }));
+        });
+
+        it('Validate failure for using a fake challenge', async () => {
+            const { proofData } = aztec.proof.bilateralSwap.constructBilateralSwap(testNotes, accounts[0]);
+
+            const fakeChallenge = new BN(crypto.randomBytes(32), 16).umod(GROUP_MODULUS).toString(10);
+
+            await exceptions.catchRevert(bilateralSwapContract.validateBilateralSwap(proofData, fakeChallenge, t2, {
+                from: accounts[0],
+                gas: 4000000,
+            }));
+        });
+
+        it('Validate failure for using fake proof data', async () => {
+            const { challenge } = aztec.proof.bilateralSwap.constructBilateralSwap(testNotes, accounts[0]);
+
+            const fakeProofData = new Array(4).map(() => new Array(6).map(() => toBytes32.randomBytes32()));
+
+            await exceptions.catchRevert(bilateralSwapContract.validateBilateralSwap(fakeProofData, challenge, t2, {
+                from: accounts[0],
+                gas: 4000000,
+            }));
+        });
+
+        it('Validate failure when points not on curve', async () => {
+            const zeroes = `${padLeft('0', 64)}`;
```

Comment:
We should construct a proof that would pass were it not for `validateCommitment` - see comments on `bilateralSwapProof.spec.js`

Reply:
Done